src/adagrams.js

```diff
@@ -1,15 +1,141 @@
+let letterDict = {
+  'A': 9,
+  'B': 2,
+  'C': 2,
+  'D': 4,
+  'E': 12,
+  'F': 2,
+  'G': 3,
+  'H': 2,
+  'I': 9,
+  'J': 1,
+  'K': 1,
+  'L': 4,
+  'M': 2,
+  'N': 6,
+  'O': 8,
+  'P': 2,
+  'Q': 1,
+  'R': 6,
+  'S': 4,
+  'T': 6,
+  'U': 4,
+  'V': 2,
+  'W': 2,
+  'X': 1,
+  'Y': 2,
+  'Z': 1
+};
+let scoreDict = {
+  'A': 1,
+  'B': 3,
+  'C': 3,
+  'D': 2,
+  'E': 1,
+  'F': 4,
+  'G': 2,
+  'H': 4,
+  'I': 1,
+  'J': 8,
+  'K': 5,
+  'L': 1,
+  'M': 3,
+  'N': 1,
+  'O': 1,
+  'P': 3,
+  'Q': 10,
+  'R': 1,
+  'S': 1,
+  'T': 1,
+  'U': 1,
+  'V': 4,
+  'W': 4,
+  'X': 8,
+  'Y': 4,
+  'Z': 10
+  };
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterList = Object.entries(letterDict)
+  let letterFreq = [];
+  for (let i = 0; i < 26; ++i) {
+    for (let j = 0; j < letterList[i][1]; ++j) {
+        letterFreq.push(letterList[i][0]);
+    }
+  }
+  let letterBank = []
+  let letterDict_deepcopy = JSON.parse(JSON.stringify(letterDict));
+  while (letterBank.length < 10){
+    let randomLetter = letterFreq[Math.floor(Math.random() * 26)];
+    if (letterDict_deepcopy[randomLetter] > 0) {
+      letterBank.push(randomLetter)
+      letterDict_deepcopy[randomLetter] -= 1
+    }
+  };
+  return letterBank;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let caseWord = input.toUpperCase();
+  for (let i of caseWord) {
+    if (!lettersInHand.includes(i)) {
+      return false;
+    } else {
+      const index = lettersInHand.indexOf(i);
+      if (index > -1) { 
+        lettersInHand.splice(index, 1); 
+      }
+    }
+  };
+  return true;
+};
+
+export const validateLetter = function charIsLetter(char) {
+  if (typeof char !== 'string') {
+    return false;
+  }
+  return /^[a-zA-Z]+$/.test(char);
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  if (word.length === 0) return 0;
+  let caseWord = word.toUpperCase();
+  let total = 0;
+  for (let i = 0; i < caseWord.length; i++){
+    if(validateLetter(word[i])== true){
+    let val = scoreDict[caseWord[i]]
+    total += val};
+  }
+  if (word.length >= 7) total += 8
+  return total;
 };
```

Comment:
I love that you are making use of a helper function and the shorthand of `if statements`! Here, you could have made use of the `for...of` loop. It would have let you access each character in the `caseWord` and indexing would have not been necessary.